src/dswx_sar/refine_with_bimodality.py

```diff
@@ -569,36 +572,34 @@ def process_dark_land_component(args):
               containing metric output
               values if debug_mode is True, otherwise, it contains zeros.
     """
-    (i, sizes, bounds, ref_land_str, landcover_str, pol_ind, bands_str,
-     water_label_str, thresholds, minimum_pixel, debug_mode) = args
+    (i, sizes, bounds, ref_land_block, landcover_block,
+     pol_ind, bands_block, water_label_block, thresholds,
+     minimum_pixel, debug_mode, startline, blocksize) = args
 
+    bounds[3] = min(bounds[3], blocksize+startline)
+    bounds[2] = max(bounds[2], startline)
     # bounding box covering the water
-    row, _, col, _ = bounds
+    x_off, _, y_off, _ = bounds
     width = bounds[1] - bounds[0]
     height = bounds[3] - bounds[2]
-    window = Window(row, col, width, height)
-
-    # Define the list of file paths
-    file_paths = [landcover_str, bands_str,
-                  ref_land_str, water_label_str]
-
-    # Initialize an empty list to store the raster datasets
-    raster_datasets = []
-
-    # Read subsets of raster datasets using a loop
-    for file_path in file_paths:
-        with rasterio.open(file_path) as src:
-            raster_subset = src.read(window=window)
-            # The method 'rasterio with window' returns 3 dimension array even
-            # though the file has 2 dimension.
-            if raster_subset.shape[0] == 1 and raster_subset.ndim == 3:
-                raster_subset = np.reshape(raster_subset,
-                                           [raster_subset.shape[1],
-                                            raster_subset.shape[2]])
-            raster_datasets.append(raster_subset)
-
-    # Assign the datasets to their respective variables
-    landcover, bands, ref_land, water_label = raster_datasets
+
+    # # Assign the datasets to their respective variables
+    # landcover, bands, ref_land, water_label = raster_datasets
+    if bands_block.ndim == 2:
+        bands_block = np.reshape(bands_block, [1,
+                                               bands_block.shape[0],
+                                               bands_block.shape[1]])
```

Comment:
More direct way of doing the same.
```suggestion
        bands_block = np.expand_dims(bands_block, axis=0)
```
Testing in terminal:
```
>>> b = np.random.rand(2,10)
>>> c = np.expand_dims(b, axis=0)
>>> c.shape
(1, 2, 10)
>>> b.shape
(2, 10)
```

Reply:
Fixed. You are right.